### PR TITLE
Simplify Kotlin DSL setup

### PR DESCRIPTION
**Pull request by wolfs**
I got quite a few warnings when trying to upgrade Gradle to 6.9.1 on the build for #530. I was wondering if using the `kotlin-dsl` plugin could work, so less manual configuration is necessary.

It seems like the compatibility class `KtDslCompatibilityUtils` was only there for Kotlin DSL versions prior to Gradle 5.0, and the plugin only supports Gradle 5.6. So I wonder if the changes I did would work and maybe simplify things.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,3 +1,19 @@
...
+
+    dependencies {
+        classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.6'
+        classpath "com.gradle.publish:plugin-publish-plugin:0.11.0"
+        classpath "com.github.ben-manes:gradle-versions-plugin:0.12.0"
+    }
+}
+
+plugins {
+    id("org.gradle.kotlin.kotlin-dsl").version("1.4.9")
```
Comment (ejona86) [inline build.gradle:14]: The typical syntax is `id "org.gradle.kotlin.kotlin-dsl" version "1.4.9"`

File: src/main/kotlin/com/google/protobuf/gradle/ProtobufConfiguratorExts.kt [hunk 2]
```diff
@@ -6,9 +6,7 @@ import org.gradle.api.Project
 import org.gradle.api.file.SourceDirectorySet
 import org.gradle.api.plugins.ExtensionAware
 import org.gradle.api.tasks.SourceSet
-import org.gradle.kotlin.dsl.NamedDomainObjectContainerScope
-import org.gradle.kotlin.dsl.closureOf
-import org.gradle.kotlin.dsl.get
+import org.gradle.kotlin.dsl.*
```
Comment (ejona86) [inline src/main/kotlin/com/google/protobuf/gradle/ProtobufConfiguratorExts.kt:9]: Why this change?

File: .github/workflows/testing.sh [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 set -eu -o pipefail
 
-export GRADLE_OPTS="-Xms128m"
+export GRADLE_OPTS="-Xms128m -Xmx512m"
```
Comment (ejona86) [inline .github/workflows/testing.sh:5]: This now needs to be reverted, since the file is now deleted.

**Review by ejona86 — COMMENTED** [review]
Overall, looks fair.

File: build.gradle [hunk 4]
```diff
@@ -93,6 +93,10 @@ artifacts {
      archives javadocJar
 }
 
+codenarc {
+    toolVersion = "1.4"
```
[the remark this answers is not in the document]
Comment (wolfs) [reply build.gradle:97]: This is the version which was used in Gradle 6.0. Gradle 6.9.1 upgrades the built-in CodeNarc version to 2.0.0. The first update was in Gradle 6.3 to 1.5: https://github.com/gradle/gradle/commit/a9d63ecdae46229a3429474373c3f386d388cded.

Even CodeNarc 1.5 fails on the sources for the rule `ImplicitClosureParameter`. I'd keep CodeNarc 1.4 now and leave the upgrade to another PR.

File: build.gradle (hunk 1 above)
Comment (wolfs) [reply build.gradle]: Changed it to the typical syntax.

File: src/main/kotlin/com/google/protobuf/gradle/ProtobufConfiguratorExts.kt (hunk 2 above)
Comment (wolfs) [reply src/main/kotlin/com/google/protobuf/gradle/ProtobufConfiguratorExts.kt]: Auto import in IDEA. Reverted to using single import statements.

File: .github/workflows/testing.sh (hunk 3 above)
Comment (wolfs) [reply .github/workflows/testing.sh]: I merged `master` and moved the change to `testing.yml`.

**Comment by wolfs** [comment]
@ejona86 Thank you for the review, I addressed your comments. Please take another look!

**Review by ejona86 — APPROVED** [review]

File: .github/workflows/testing.sh (hunk 3 above)
Comment (ejona86) [reply .github/workflows/testing.sh]: Hmm... The commit messages for this change is strange message, as it doesn't match the change ta all: 17ffe1d4f9690311a814c85f90d333ea7de8afd0

I had assumed that the max heap size of 512 MB was necessary to fit within the runner's memory limits. But that message just confuses things.

Comment (wolfs) [reply .github/workflows/testing.sh]: Yeah, re-reading the commit message I can see that it is misleading. The problem is that Gradle 6.9 adds `-Xmx64` as a Gradle option automatically. That means you can't use `-Xms128m`, since then the minimum heap would be bigger as the maximum heap which fails the java executable. Probably no Java options are necessary to run the build at all, let me try.

Comment (wolfs) [reply .github/workflows/testing.sh]: I removed the `GRADLE_OPTS` and the build passes. Is everything good to go now?

Comment (ejona86) [reply .github/workflows/testing.sh]: Ooh, I see. it did say _minimum_ was 64 MB. Interesting. That's small, but also makes it clear the type of memory usage that is acceptable ("If you're big, use another process!").

Yeah, I'm fine removing them instead. Obviously, things are different now with how Gradle uses memory.
